examples/ecs/change_detection.rs

```diff
@@ -0,0 +1,106 @@
+//! This example illustrates how to react to component and resource changes.
+
+use bevy::prelude::*;
+use rand::Rng;
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_systems(Startup, setup)
+        .add_systems(
+            Update,
+            (
+                change_component,
+                change_component_2,
+                change_resource,
+                change_detection,
+            ),
+        )
+        .run();
+}
+
+#[derive(Component, PartialEq, Debug)]
+struct MyComponent(f32);
+
+#[derive(Resource, PartialEq, Debug)]
+struct MyResource(f32);
+
+fn setup(mut commands: Commands) {
+    // Note the first change detection log correctly points to this line because the component is
+    // added. Although commands are deferred, they are able to track the original calling location.
+    commands.spawn(MyComponent(0.0));
+    commands.insert_resource(MyResource(0.0))
```

Comment:
```suggestion
    commands.insert_resource(MyResource(0.0));
```

Clippy wants a semicolon here, but that should make CI pass.